src/main/java/Task.java

```diff
@@ -0,0 +1,60 @@
+public abstract class Task {
+    protected String name;
+    protected boolean isDone;
+
+    /**
+     * Constructs a new Task with the given name and sets its completion status to false.
+     *
+     * @param name The name of the task.
+     */
+    public Task(String name) {
+        this.name = name;
+        this.isDone = false;
+    }
+
+    public static Task fromFileFormat(String line) throws DukeException {
```

Comment:
Should this method have a javadoc?

Reply:
Thanks! I agree that it should have additional documentation due to its complexity and I've added the JavaDoc